core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/pushpull/LDAPMembershipPullActions.java

```diff
@@ -141,7 +141,7 @@ protected List<Object> getMembAttrValues(final SyncDelta delta, final Connector
      */
     @Transactional(readOnly = true)
     @Override
-    public void beforeUpdate(
+    public synchronized void beforeUpdate(
```

Comment:
please find a different way to solve the issue which does not involve synchronizing the whole methods

Reply:
yes, use ConcurrentHashMap is better way.